src/main/java/io/permit/sdk/Permit.java

```diff
@@ -104,10 +105,11 @@ public Permit(PermitConfig config) {
      * @param resource The resource object representing the resource.
      * @param context  The context object representing the context in which the action is performed.
      * @return {@code true} if the user is authorized, {@code false} otherwise.
-     * @throws IOException if an error occurs while checking the authorization.
+     * @throws PermitApiError if an error occurs while sending the authorization request to the PDP.
+     * @throws IOException if could not read the content of the returned http response.
      */
     @Override
-    public boolean check(User user, String action, Resource resource, Context context) throws IOException {
+    public boolean check(User user, String action, Resource resource, Context context) throws IOException, PermitApiError {
```

Comment:
Not an expert with Java but isn't this a breaking change to add another type of throwing exception ? ( the calling function would also need to do so )

Reply:
you're right - it's a small (necessary) change but it is breaking.
i will bump the version to 2.0.0